### PR TITLE
feat: Renderer process ANR detection with stack traces

### PR DESCRIPTION
**Pull request by timfish**
When https://github.com/getsentry/sentry-javascript/pull/9191 is merged, `anr-utils.ts` in this PR can be removed.

This PR adds ANR detection for the Electron renderer processes.

This is entirely configured from each renderer processes:
```ts
import { init } fron '@sentry/electron/renderer';

init({
  dsn: '__DSN__',
  anrDetection: { pollInterval: 200, anrThreshold: 2000, captureStackTrace: true },
})
```
Renderer ANR has no overhead unless a renderer starts sending ANR status updates to the main process.

### CODE REVIEW

File: src/renderer/sdk.ts [hunk 1]
```diff
@@ -7,20 +7,25 @@ import {
...
 import { ensureProcess } from '../common';
+import { enableAnrRendererMessages } from './anr';
 import { ScopeToMain } from './integrations';
 import { electronRendererStackParser } from './stack-parse';
 import { makeRendererTransport } from './transport';
 
 export const defaultIntegrations = [...defaultBrowserIntegrations, new ScopeToMain()];
 
+interface ElectronRendererOptions extends BrowserOptions {
+  anr?: boolean;
```
Comment (AbhiPrasad) [inline src/renderer/sdk.ts:18]: let's make this variable name more descriptive, like `enableAnrDetection`.

**Review by AbhiPrasad — COMMENTED** [review]
I'm in favor of turning it on by default 👍

**Comment by AbhiPrasad** [comment]
anr utils changes released with https://github.com/getsentry/sentry-javascript/releases/tag/7.74.0

**Comment by timfish** [comment]
@AbhiPrasad 

Done, what do you think about the above mentioned renaming of `enableAnrDetection` to `enableMainProcessAnrDetection` and removing the `mainProcess` config option?

**Comment by AbhiPrasad** [comment]
Yeah I think let's use `enableMainProcessAnrDetection` here, makes it behavior very clear.

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**